### ISSUE TITLE
TsConfigPathsPlugin setting configFileName issue

### ISSUE THREAD

**Issue by rimlin** [s-panferov/awesome-typescript-loader#463]
Setting property `configFileName` didn't have any effect at end point path of tsconfig:
Example of config:
```
...
plugins: [
      new TsConfigPathsPlugin({
        configFileName: 'src\\tsconfig.server.json'
      }),
    ],
...

```

Result object of webpack config in console:
```
 plugins:  [
...
     PathPlugin {
       source: 'described-resolve',
       target: 'resolve',
       ts: [Object],
       options: [Object],
       configFilePath: 'D:\\project\\src\\tsconfig.server.json',
       baseUrl: undefined,
       absoluteBaseUrl: 'D:\\project\\src',
       mappings: [] },

     ....
]
```

Message from loader:

```
[at-loader] Using typescript@2.3.2 from typescript and "tsconfig.json" from D:\project/tsconfig.json.
```

Enviroment:

OS: Window 10
Webpack: 2.2.1
typescript: 2.2.1
awesome-typescript-loader: 3.2.1

**Comment by nasreddineskandrani** [comment]
did you add plugins to the resolve section of webpack?

**Comment by Delagen** [comment]
@rimlin try to specify the same option to query of loader
```
 {
test:    /\.ts$/,
loader:"awesome-typescript-loader",
query:{
configFileName: 'src\\tsconfig.server.json'
}
}
```

**Comment by hstarorg** [comment]
@Delagen That's  very hard for use. `TsConfigPathsPlugin` is  used for set different config file, but if config it in loader query,I need config rules in any files.

**Comment by Delagen** [comment]
@hstarorg I don't think that is difficult to specify global query in webpack.config.js

**Comment by hstarorg** [comment]
@Delagen In my project, I have some tsconfig files, so I must set it in `opt.modules.rules[0].use[0].options = {configFileName: 'tsconfig path'}`, it's depend my rule order.

In the document, it says we can use `TsConfigPathsPlugin` to set configFileName, but in fact it is not.

**Comment by Delagen** [comment]
@hstarorg It depends how you generate your webconfig. I generate it dynamically, so I simply paste it where it needed ))

**Comment by hstarorg** [comment]
Yes, we can have solution to resolve it. But the `TsConfigPathsPlugin`'s effect is not equal the document.
